### PR TITLE
Working indexing from SQS for Review

### PR DESCRIPTION
**Pull request by alexgleith**
This indexes from an SQS queue.

Seems to work ok in the one scenario I've tested in.

It doesn't do any fancy iteration, but I think it will provide adequate performance for realtime indexing. The intention would be that this runs once every 1/3/12/24 hours and gets all the items off a queue.

It can be tested in the us-west-2 region in the DE Africa account with the `test_africa` queue.

### CODE REVIEW

File: odc_index/sqs_to_dc.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    help="Expect STAC 1.0 metadata and attempt to transform to ODC EO3 metadata",
+)
+@click.option(
+    "--limit",
+    default=None,
+    type=int,
+    help="Stop indexing after n datasets have been indexed.",
+)
+@click.argument("queue_name", type=str, nargs=1)
+@click.argument("product", type=str, nargs=-1)
```
Comment (whatnick) [inline odc_index/sqs_to_dc.py:101]: If you are planning to pass multiple products to match to it better to pass this as a quotes string and a single parameter to be more k8s executor friendly. Then split them into products in code from the input string. See Thredds-to-dc for a sample.

File: README.rst [hunk 2]
```diff
@@ -14,6 +14,7 @@ These utilities include:
 
 #. **s3-to-dc** : Index from S3 storage to a Datacube database.
 #. **thredds-to-dc** : Index from Thredds server to a Datacube database.
+#. **sqs-to-dc** : Index from SQS queue to a Datacube database.
```
Comment (whatnick) [inline README.rst:17]: Please create an issue and tag some commits to the issue to help in writing release notes.

File: odc_index/sqs_to_dc.py (hunk 1 above)
Comment (alexgleith) [reply odc_index/sqs_to_dc.py:101]: I copied this from the other `s3_to_dc` script.

This is the code in the thredds one: `@click.argument("product", type=str, nargs=1)
` and I don't see how it's different, aside from having `-1` nargs...

**Comment by alexgleith** [comment]
Closes #54

File: odc_index/sqs_to_dc.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+) -> Tuple[int, int]:
+
+    ds_added = 0
+    ds_failed = 0
+
+    queue_empty = False
+
+    while not queue_empty and (not limit or ds_added + ds_failed < limit):
+        messages = queue.receive_messages(
+            VisibilityTimeout=30, MaxNumberOfMessages=1, MessageAttributeNames=["All"]
```
Comment (tom-butler) [inline odc_index/sqs_to_dc.py:27]: Should set WaitTimeSeconds to ensure the queue is _actually_ empty, and not still being filled.

Comment (alexgleith) [reply odc_index/sqs_to_dc.py]: Done

**Pull request merged**